### ISSUE TITLE
urlSync bug with base attribute

### ISSUE THREAD

**Issue by alexandremeunier** [algolia/instantsearch.js#790]
Hello,

There is a bug happening when `urlSync: true` and the html page has a `base` attribute which is different from the current URL. It will instantly replace the URL with the `base` url, to which is appended the instantsearch.js state as a query string

Demo: http://url-syncbug.surge.sh/test

The base attribute is: 
```html
<base href="http://url-syncbug.surge.sh">
```

The page URL is instantly changed to http://url-syncbug.surge.sh/?q=rake&hPP=6&idx=Version_development&p=0&is_v=1

One solution could be to replace the entire URI path instead of just the query params part

**Comment by vvo** [comment]
Working on this, got a fix (use the full url). The underlying issue is that as soon as you use a base href,  when you push relative things inside pushState or location.href then they are resolved to the base href tag.

Which can either:
- result in a full page load (when using #)
- result in a security issue (when using pushState)

**Comment by GabrielDelepine** [comment]
This is my fix (really really durty)

Be carefull, it's durty.

...

Did I told you it's durty ? Yes ?

```javascript
        instantsearch.widgets.durtyFixPagination = function() {
            return {
                render: function render(options) {

                    setTimeout(function() {
                        var matches = document.querySelectorAll("#pagination a[href]");
                        var domain = (qpConfig2.WEBAPPMOBILE ? qpConfig2.URLS.MOBILE : qpConfig2.URLS.WWW);
                        var urlBegin = domain + '/?';
                        angular.forEach(matches, function(a) {
                            if (urlBegin === a.href.substr(0, urlBegin.length)) {
                                a.href = domain + qpRoute.getCurrentRoute().translatedUrl + a.href.substr(urlBegin.length - 1);
                            }
                        });
                        console.info('pagination fix done');
                    }, 800);

                }
            };
        };
```

**Comment by vvo** [comment]
@GabrielDelepine this should already be fixed in latest version, wonder why did you had to fix it? Are you on latest version?

**Comment by GabrielDelepine** [comment]
Hi @vvo 

The fix is in the master branch. I am waiting for the release ```1.4.3```.

**Comment by vvo** [comment]
@GabrielDelepine The fix was done and is available in 1.4.2, that's why I am wondering. See https://github.com/algolia/instantsearch.js/commit/e58aadcd9aae77165b802b46a7d3de5864569e1e

**Comment by vvo** [comment]
Please open a new issue if you still have problems with urlSync and base tag
